CIP-????/README.md

```diff
@@ -0,0 +1,72 @@
+Hyperlane is a permissionless interoperability layer that facilitates communication of arbitrary data between smart contracts across multiple blockchains. Hyperlane's [interchain security modules](https://docs.hyperlane.xyz/docs/protocol/sovereign-consensus)
+rely on the verification of specific cryptographic proofs from one chain to another. These proofs utilize the `keccak256` hash to calculate consistent cross-chain message IDs.
+The multi-signature module verifies that a majority of off-chain validators have signed an ECDSA signature over a `keccak256` digest, a common practice in EVM.
+
+While Hyperlane [can support](https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2399) different cryptographic primitives for non-EVM chains, doing so could compromise censorship resistance, resulting in only limited support for Cardano in Hyperlane. By implementing this CIP, Cardano could fully integrate with Hyperlane's security modules, enabling Cardano smart contracts to communicate with any blockchain supported by Hyperlane.
+
+## Specification
+This proposal aims to introduce a new built-in hash function `keccak_256`.
+
+This function will be developed following the [`keccak256`](https://keccak.team/files/Keccak-submission-3.pdf) specification 
+and will utilize the [cryptonite](https://github.com/haskell-crypto/cryptonite/blob/master/Crypto/Hash/Keccak.hs) implementation. 
+Since `cryptonite` is already a part of the [`cardano-base`](https://github.com/input-output-hk/cardano-base/blob/master/cardano-crypto-class/src/Cardano/Crypto/Hash/Keccak256.hs), 
+this simplifies its integration into Plutus. The cost of the `keccak_256` operation will scale linearly with the length of the message.
+
+More specifically, Plutus will gain the following primitive operation:
+
+* `keccak_256` :: ByteString -> ByteString
+
+The input to this function can be a `ByteString` of arbitrary size, and the output will be a `ByteString` of 32 bytes. 
+Note that this function aligns with the format of existing hash functions in Plutus, such as [blake2b_256](https://github.com/input-output-hk/plutus/blob/75267027f157f1312964e7126280920d1245c52d/plutus-core/plutus-core/src/Data/ByteString/Hash.hs#L25)
+
+## Rationale: how does this CIP achieve its goals?
+While the `keccak256` function might be implemented in on-chain scripts, doing so would be computationally unfeasible. 
+
+## Path to Active
+This CIP may transition to active status once the Plutus version containing the `keccak_256` function is introduced 
+in a node release and becomes available on Mainnet.
+
+### Acceptance Criteria
+* A Plutus binding is created for the `keccak256` function and included in a new version of Plutus.
+* Integration tests, similar to those of the existing Plutus hash functions, are added to the testing infrastructure.
+* The function is benchmarked to assess its cost.
```

Comment:
```suggestion
* The function is benchmarked to assess its cost. As for other hash functions available in Plutus (blake2b and sha256), we expect the cost of keccak to be linear with respect to the size of the message.
```

Reply:
@rphair . I've suggested the comment that we expect the cost of the function to be linear wrt the message size as part of the AC. Do you think it should go somewhere else?

Reply:
@iquerejeta it looks sufficient to me there.  @michaelpj I don't know if a more detailed cost analysis is required in the CIP itself as per cip35... I was struggling to hold onto any concrete requirements from the meeting so maybe @KtorZ @SebastienGllmt could better answer this.

Reply:
Resolved in https://github.com/cardano-foundation/CIPs/pull/524/commits/b0dc15e3cc43c6486c3656ac621991aa9ea87d21.